lib/components/viewers/route-viewer.js

```diff
@@ -242,72 +340,73 @@ const RouteNameElement = styled(Label)`
+       route, scroll to it. The initialRender prop prohibits the row being scrolled to
+       if the user has clicked on a route
+    */
+    if (this.props.isActive && this.props.initialRender) {
+      this.activeRef.current.scrollIntoView()
+    }
+  }
+
   _onClick = () => {
-    const { findRoute, isActive, route, setViewedRoute } = this.props
+    const { findRoute, getVehiclePositionsForRoute, isActive, route, setViewedRoute } = this.props
     if (isActive) {
       // Deselect current route if active.
-      setViewedRoute({ routeId: null })
+      setViewedRoute({ patternId: null, routeId: null })
     } else {
       // Otherwise, set active and fetch route patterns.
       findRoute({ routeId: route.id })
+      getVehiclePositionsForRoute(route.id)
       setViewedRoute({ routeId: route.id })
     }
   }
 
-  getCleanRouteLongName ({ longNameSplitter, route }) {
-    let longName = ''
-    if (route.longName) {
-      // Attempt to split route name if splitter is defined for operator (to
-      // remove short name value from start of long name value).
-      const nameParts = route.longName.split(longNameSplitter)
-      longName =
-        longNameSplitter && nameParts.length > 1
-          ? nameParts[1]
-          : route.longName
-      // If long name and short name are identical, set long name to be an empty
-      // string.
-      if (longName === route.shortName) longName = ''
-    }
-    return longName
-  }
-
   render () {
     const { isActive, operator, route } = this.props
     const { ModeIcon } = this.context
 
-    const { defaultRouteColor, defaultRouteTextColor, longNameSplitter } =
-      operator || {}
-    const backgroundColor = `#${defaultRouteColor || route.color || 'ffffff'}`
-    // NOTE: text color is not a part of short response route object, so there
-    // is no way to determine from OTP what the text color should be if the
-    // background color is, say, black. Instead, determine the appropriate
-    // contrast color and use that if no text color is available.
-    const contrastColor = getContrastYIQ(backgroundColor)
-    const color = `#${defaultRouteTextColor ||
-      route.textColor ||
-      contrastColor}`
-    // Default long name is empty string (long name is an optional GTFS value).
-    const longName = this.getCleanRouteLongName({ longNameSplitter, route })
+    const { backgroundColor, color, longName } = getColorAndNameFromRoute(
+      operator,
+      route
+    )
+
     return (
-      <StyledRouteRow isActive={isActive}>
+      <StyledRouteRow isActive={isActive} ref={this.activeRef} routeColor={backgroundColor}>
         <RouteRowButton
           className='clear-button-formatting'
+          isActive={isActive}
           onClick={this._onClick}
+          routeColor={backgroundColor}
         >
           <RouteRowElement>
             {operator && operator.logo && (
               <OperatorImg alt={`${operator.name} logo`} src={operator.logo} />
```

Comment:
This string may need to be localized.

Reply:
Localized in 25aa7249aea9a0b68a5b47e9c93e3a94490dc25b